app/helpers/catalog_helper.rb

```diff
@@ -86,6 +86,7 @@ def date_list( document )
         end
       end
     end
+    dateVal = dateVal[0,dateVal.length()-2] if !dateVal.blank? && dateVal.end_with?(', ')

```

Comment:
This looks overly complex too me, and would still not handle dateVal being nil.  How about:

```
dateVal = dateVal.gsub(/, $/,'') if dateVal
```